stock-tracker/src/features/keyStats/redux/keyStatsReducer.test.ts

```diff
@@ -0,0 +1,60 @@
+import { keyStatsReducer, KeyStatsState } from "./keyStatsReducer";
+import { SET_RESPONSE, SET_ERROR_KEYSTATS } from "./actionTypes";
+import { ADD_SYMBOL } from "../../search/redux/actionTypes";
+import { SetSymbol } from "../../search/redux/actions";
+import { Response, KeyStatsActions } from "./actions";
+
+describe("testing keyStats reducer", () => {
+  let initialState: KeyStatsState = {
+    response: undefined,
+    loading: false,
+    error: false
+  };
+
+  let newState: KeyStatsState;
+
+  describe("returns loading as true after the ADD_SYMBOL action is called", () => {
+    const action: SetSymbol = {
+      type: ADD_SYMBOL,
+      payload: "AAPL"
+    };
+
+    beforeAll(() => {
+      newState = keyStatsReducer(initialState, action);
+    });
+
+    it("should update the state", () => {
+      expect(newState.loading).toBeTruthy();
+    });
+  });
+
+  describe("returns expected payload when SET_RESPONSE action is called", () => {
+    const payload = {} as Response;
```

Comment:
q: This looks a bit strange to me, why are you type casting the `payload`?

Reply:
I tried it here differently since the response object is so large and I might not want to type it all out for testing